app/models/share/button.rb

```diff
@@ -24,4 +24,6 @@ def share_progress?
     # the share types currently managed by share progress
     %w[facebook twitter email].include? sp_type
   end
+
+  scope :sp, -> { where sp_type: %w[facebook twitter email] }
```

Comment:
Could we name the scope `share_progress`? It's easier to understand that than `sp` (it took me a few seconds think share progress).